src/libcore/convert.rs

```diff
@@ -48,7 +48,23 @@
 
 #![stable(feature = "rust1", since = "1.0.0")]
 
-use str::FromStr;
+use fmt;
+
+/// A type used as the error type for implementations of fallible conversion
+/// traits in cases where conversions cannot actually fail.
+///
+/// Because `Infallible` has no variants, a value of this type can never exist.
+/// It is used only to satisfy trait signatures that expect an error type, and
+/// signals to both the compiler and the user that the error case is impossible.
+#[unstable(feature = "try_from", issue = "33417")]
+pub enum Infallible {}
```

Comment:
I'm still too not sold on the name `Infalliable`; perhaps `NeverError` or something similar would be better?

Reply:
We also should have more impls for this, like `Eq`, `PartialEq`, `Ord`, `PartialOrd`, `Hash`, and `Error`.

Reply:
Agreed on the extra derived impls. I will add those. `NeverError` probably goes against the style guidelines around "stutter" in names. I'd be open to just `Never`, though. After all, that's the way you say `!` (as a type).

Reply:
After thinking about it, maybe `CannotError`?  I think it should probably end in `Error` by convention.